pkg/volume/csi/nodeinfomanager/nodeinfomanager_test.go

```diff
@@ -637,6 +637,110 @@ func TestInstallCSIDriverExistingAnnotation(t *testing.T) {
 	}
 }
 
+func TestValidateCSINodeInfo(t *testing.T) {
+	testcases := []struct {
+		name      string
+		nodeInfo  *csiv1alpha1.CSINodeInfo
+		expectErr bool
+	}{
+		{
+			name: "multiple drivers with same ids and different topology keys",
+			nodeInfo: &csiv1alpha1.CSINodeInfo{
+				CSIDrivers: []csiv1alpha1.CSIDriverInfo{
+					{
+						Driver:       "driver1",
+						NodeID:       "node1",
+						TopologyKeys: []string{"key1, key2"},
+					},
+					{
+						Driver:       "driverB",
+						NodeID:       "nodeA",
+						TopologyKeys: []string{"keyA", "keyB"},
+					},
+				},
+			},
+			expectErr: false,
+		},
+		{
+			name: "multiple drivers with same ids and similar topology keys",
+			nodeInfo: &csiv1alpha1.CSINodeInfo{
+				CSIDrivers: []csiv1alpha1.CSIDriverInfo{
+					{
+						Driver:       "driver1",
+						NodeID:       "node1",
+						TopologyKeys: []string{"key1"},
+					},
+					{
+						Driver:       "driver2",
+						NodeID:       "node1",
+						TopologyKeys: []string{"key1"},
+					},
+				},
+			},
+			expectErr: false,
+		},
+		{
+			name: "duplicate drivers",
+			nodeInfo: &csiv1alpha1.CSINodeInfo{
+				CSIDrivers: []csiv1alpha1.CSIDriverInfo{
+					{
+						Driver:       "driver1",
+						NodeID:       "node1",
+						TopologyKeys: []string{"key1", "key2"},
+					},
+					{
+						Driver:       "driver1",
+						NodeID:       "nodeX",
+						TopologyKeys: []string{"keyA", "keyB"},
+					},
+				},
+			},
+			expectErr: true,
+		},
+		{
+			name: "single driver with duplicate topology keys",
```

Comment:
Did you also verify that when there's both a driver and topology conflict, both errors get printed? I don't think there needs to be a separate test case for that but just wanted to make sure

Reply:
Yup. Example: ```duplicate entries found for driver: driver1 in driver statuses, duplicate entries found for driver: driver1 in driver specs```